keyboards/crkbd/keymaps/gotham/rgb.c

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#ifdef RGBLIGHT_ENABLE
+  extern rgblight_config_t rgblight_config;
+  rgblight_config_t RGB_current_config;
+#endif
+
+#ifdef RGB_MATRIX_ENABLE
+  extern rgb_matrix_config_t rgb_matrix_config;
+  rgb_config_t RGB_current_config;
+#endif
+
+
+void save_rgb_config(void) {
+  #ifdef RGBLIGHT_ENABLE
+    RGB_current_config.enable = rgblight_config.enable;
+    RGB_current_config.mode = rgblight_config.mode;
+    RGB_current_config.speed  = rgblight_config.speed;
+    RGB_current_config.hue = rgblight_config.hue;
+    RGB_current_config.sat = rgblight_config.sat;
+    RGB_current_config.val = rgblight_config.val;
```

Comment:
Most of these can be retrieved by a function:
https://github.com/qmk/qmk_firmware/blob/master/quantum/rgblight.h#L297-L305